common/lib/xmodule/xmodule/library_root_xblock.py

```diff
@@ -114,3 +132,12 @@ def parse_xml(cls, xml_data, system, id_generator, **kwargs):
     def add_xml_to_node(self, resource_fs):
         """ XML support not yet implemented. """
         raise NotImplementedError
+
+    @XBlock.json_handler
+    def trigger_previews(self, request_body, suffix):  # pylint: disable=unused-argument
+        """ Enable or disable previews in studio for library children. """
+        self.show_children_previews = request_body.get('show_children_previews', self.show_children_previews)
+        user_id = self.runtime.service(self, 'user').get_current_user().opt_attrs['edx-platform.user_id']
+        # pylint: disable=no-member
+        self.system.modulestore.update_item(self, user_id)
```

Comment:
The preceding three lines break the API abstractions and are not actually necessary as far as I can tell. The runtime will always save the block after calling any handler so there is no need to call `update_item` manually.

Also, I'm still not sure it makes more sense to store this state in an XBlock field and not just a browser cookie. But I think we've had that discussion elsewhere already.